go/vt/mysqlctl/s3backupstorage/s3.go

```diff
@@ -170,16 +172,20 @@ func (bh *S3BackupHandle) AddFile(ctx context.Context, filename string, filesize
 			u.PartSize = partSizeBytes
 		})
 		object := objName(bh.dir, bh.name, filename)
-
-		_, err := uploader.Upload(&s3manager.UploadInput{
+		sendStats := bh.bs.params.Stats.Scope(stats.Operation("AWS:Request:Send"))
+		_, err := uploader.UploadWithContext(ctx, &s3manager.UploadInput{
```

Comment:
Not fully sure of the implications of switching to `UploadWithContext`.

Reply:
What motivated you to make the change in the first place?
`ctx` will have a timeout associated with it. Assuming standard practice, `UploadWithContext` will terminate and return if that timeout is exceeded.

Reply:
Motivation was that I _thought_ `UploadWithContext` let me specify a `WithUploaderRequestOptions`, whereas `Upload` does not, but now I can see that's not the case, so I have no reason to use this.

Maybe specifying a context would be a good thing, but I don't know enough to say that with any confidence. Reverted.

Reply:
Either you should change this back to be consistent with the change in Line 222 (GetObjectWithContext), or change that back to not use a context. I vote for using a context.

Reply:
Oops didn't even notice it was in two places when I tried switching back 😮‍💨 

Will switch back to `GetObjectWithContext`